packages/flutter/test/widgets/image_test.dart

```diff
@@ -35,6 +35,57 @@ void main() {
     imageCache.maximumSize = originalCacheSize;
   });
 
+  testWidgets('Verify Image does not use disposed handles', (WidgetTester tester) async {
+    final ValueNotifier<int> outerListenable = ValueNotifier<int>(0);
+    final ValueNotifier<int> innerListenable = ValueNotifier<int>(0);
+
+    bool imageLoaded = false;
+    MemoryImage image = MemoryImage(Uint8List.fromList(kTransparentImage));
+
+    Future<void> runAsyncAndIdle() async {
+      for (int i = 0; i < 20; ++i) {
+        await tester.runAsync(() => Future<void>.delayed(Duration.zero));
+        await tester.idle();
+      }
+    }
+
+    await tester.pumpWidget(ValueListenableBuilder<int>(
+      valueListenable: outerListenable,
+      builder: (BuildContext _, Object? __, Widget? ___) => Image(
+        image: image,
+        frameBuilder: (BuildContext _, Widget child, int? __, bool ___) {
+          if (((child as Semantics).child! as RawImage).image != null) {
+            imageLoaded = true;
+          }
+          return LayoutBuilder(
+            builder: (BuildContext _, BoxConstraints __) => ValueListenableBuilder<int>(
+              valueListenable: innerListenable,
+              builder: (BuildContext _, Object? __, Widget? ___) => KeyedSubtree(
+                key: UniqueKey(),
+                child: child,
+              ),
+            ),
+          );
+        },
+      ),
+    ));
+
+    while (!imageLoaded) {
+      await runAsyncAndIdle();
+      await tester.pump();
+    }
+
+    image = MemoryImage(Uint8List.fromList(kBlueSquarePng));
+    outerListenable.value++;
```

Comment:
nit: Flutter style says do not use `++` or `--`. 
```suggestion
    outerListenable.value += 1;
```

(And below).